cmd/configurator/main.go

```diff
@@ -0,0 +1,316 @@
+								Path: "/stats/summary",
+								Port: intstr.IntOrString{
+									Type:   intstr.String,
+									StrVal: "metrics",
+								},
+							},
+						},
+					},
+					Env: []v1.EnvVar{
+						{
+							Name: "VK_HOST_NAME",
+							ValueFrom: &v1.EnvVarSource{
+								FieldRef: &v1.ObjectFieldSelector{
+									FieldPath: "spec.nodeName",
+								},
+							},
+						},
+						{
+							Name: "VK_POD_NAME",
+							ValueFrom: &v1.EnvVarSource{
+								FieldRef: &v1.ObjectFieldSelector{
+									FieldPath: "metadata.name",
+								},
+							},
+						},
+						{
+							Name: "VKUBELET_POD_IP",
+							ValueFrom: &v1.EnvVarSource{
+								FieldRef: &v1.ObjectFieldSelector{
+									FieldPath: "status.podIP",
+								},
+							},
+						},
+						{
+							Name:  "PARTITION",
+							Value: partitionName,
+						},
+						{
+							Name:  "RED_BOX_SOCK",
+							Value: *redBoxSock,
+						},
+						{
+							Name:  "APISERVER_CERT_LOCATION",
+							Value: "/kubelet.crt",
+						},
+						{
+							Name:  "APISERVER_KEY_LOCATION",
+							Value: "/kubelet.key",
+						},
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "syslurm-mount",
+							MountPath: "/syslurm",
+						},
+						{
+							Name:      "kubelet-crt",
+							MountPath: "/kubelet.crt",
+						},
+						{
+							Name:      "kubelet-key",
+							MountPath: "/kubelet.key",
+						},
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: "syslurm-mount", // directory with red-box socket
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
+							Path: "/var/run/syslurm",
+							Type: &[]v1.HostPathType{v1.HostPathDirectory}[0],
+						},
+					},
+				},
+				{
+					Name: "kubelet-crt", // we need certificates for pod rest api, k8s gets pods logs via rest api
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
+							Path: "/var/lib/kubelet/pki/kubelet.crt",
+							Type: &[]v1.HostPathType{v1.HostPathFile}[0],
+						},
+					},
+				},
+				{
+					Name: "kubelet-key",
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
+							Path: "/var/lib/kubelet/pki/kubelet.key",
+							Type: &[]v1.HostPathType{v1.HostPathFile}[0],
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// partitionNames extracts slurm partition name from k8s node labels
+func partitionNames(nodes []v1.Node) []string {
+	names := make([]string, 0, 0)
+	for _, n := range nodes {
+		if l, ok := n.Labels["slurm.sylabs.io/partition"]; ok {
+			names = append(names, l)
+		}
+	}
+
+	return names
+}
+
+// notIn returns elements from s1 which are not presented in s2
+func notIn(s1, s2 []string) []string {
+	notIn := make([]string, 0, 0)
+	for _, e1 := range s1 {
+		if contains(e1, s2) {
+			continue
+		}
+		notIn = append(notIn, e1)
+	}
+
+	return notIn
+}
+
+// contains checks if elements presents in s
+func contains(e string, s []string) bool {
```

Comment:
```suggestion
func contains(s []string, e string) bool {
```

Usually first parameter is against what operation is performed, e.g https://golang.org/pkg/strings/#Contains